src/pygeoprocessing/routing/routing.pyx

```diff
@@ -409,9 +439,21 @@ cdef class _ManagedRaster:
             <int>block_index, <double*>double_buffer, removed_value_list)
 
         if self.write_mode:
-            raster = gdal.OpenEx(
-                self.raster_path, gdal.GA_Update | gdal.OF_RASTER)
-            raster_band = raster.GetRasterBand(self.band_id)
+            n_attempts = 5
+            while True:
+                raster = gdal.OpenEx(
+                    self.raster_path, gdal.GA_Update | gdal.OF_RASTER)
+                if raster is None:
+                    if n_attempts == 0:
+                        raise RuntimeError(
+                            f'could not open {self.raster_path} for writing')
+                    LOGGER.warning(
+                        f'opening {self.raster_path} resulted in null, '
+                        f'trying {n_attempts} more times.')
+                    n_attempts -= 1
+                    time.sleep(0.5)
+                raster_band = raster.GetRasterBand(self.band_id)
+                break

```

Comment:
It came up when testing the subwatershed routing... that's not to say i wouldn't want it everywhere else but it was specifically happening during this development. What I'd *really* like is to avoid duplicating this class in every Cython module, but every time I try to abstract this into a Cython class things start to break... I think bevcause of the linkage to the LRUCache.h file. I think we've both attempted it over the years. Can I punt on duplicating this code everywhere else and if it's an issue in the future we try really hard to get a ManagedRaster abstract component we can use in all the pygeoprocessing cython modules?

Reply:
Oh for sure!  We've definitely both tried to abstract this out, but it became clear that it's going to take a C++ extension to really do it right with templating.  So yes, let's punt.